google/resource_dataproc_cluster.go

```diff
@@ -0,0 +1,870 @@
+
+					if len(value) > 55 {
+						errors = append(errors, fmt.Errorf(
+							"%q cannot be longer than 55 characters", k))
+					}
+					if !regexp.MustCompile("^[a-z0-9-]+$").MatchString(value) {
+						errors = append(errors, fmt.Errorf(
+							"%q can only contain lowercase letters, numbers and hyphens", k))
+					}
+					if !regexp.MustCompile("^[a-z]").MatchString(value) {
+						errors = append(errors, fmt.Errorf(
+							"%q must start with a letter", k))
+					}
+					if !regexp.MustCompile("[a-z0-9]$").MatchString(value) {
+						errors = append(errors, fmt.Errorf(
+							"%q must end with a number or a letter", k))
+					}
+					return
+				},
+			},
+
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "global",
+				ForceNew: true,
+			},
+
+			"zone": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"delete_autogen_bucket": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: false,
+			},
+
+			"staging_bucket": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"bucket": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+
+			"image_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"network": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"subnetwork"},
+				StateFunc: func(s interface{}) string {
+					return extractLastResourceFromUri(s.(string))
+				},
+			},
+
+			"subnetwork": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"network"},
+				StateFunc: func(s interface{}) string {
+					return extractLastResourceFromUri(s.(string))
+				},
+			},
+
+			"labels": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				ForceNew: false,
+				Elem:     schema.TypeString,
+			},
+
+			"properties": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				ForceNew: false,
+				Elem:     schema.TypeString,
+			},
+
+			"tags": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"master_config": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"num_masters": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+							ForceNew: false,
+						},
+
+						"machine_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ForceNew: true,
+						},
+
+						"num_local_ssds": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
```

Comment:
Technically no. I had originally added it as computed to ensure it was listed as output for people to potentially use should they want to use terraform to build up a picture of what the cluster is comprised of, but this is probably overkill, removed.